common/cpp/crypto/mbedtls/skenc.cpp

```diff
@@ -0,0 +1,240 @@
+/* Copyright 2018-2020 Intel Corporation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * @file
+ * Secret key (symmetric) encryption.
+ * Uses AES-GCM 256, which also includes authentication.
+ *
+ * Lower-level functions implemented using MBed TLS.
+ * See also skenc_common.cpp for MBed TLS-independent code.
+ */
+
+#include <string.h>  // memcmp()
+#include <mbedtls/gcm.h>
+
+#include "crypto_shared.h"
+#include "error.h"
+#include "skenc.h"
+
+#ifndef CRYPTOLIB_MBEDTLS
+#error "CRYPTOLIB_MBEDTLS must be defined to compile source with Mbed TLS."
+#endif
```

Comment:
It is defined in your environment. From `common/cpp/tests/Makefile`:

```
# To use Mbed TLS instead of the default OpenSSL, define CRYPTOLIB_MBEDTLS=1
# For example: CRYPTOLIB_MBEDTLS=1 make
```

In the future, MBed TLS may become the default (the default is defined in `common/cpp/crypto/crypto_shared.h` )